### PR TITLE
utils: use safer parsing utility across codebase

### PR DESCRIPTION
**Pull request by rohityadavcloud**
This addresses SonarQube/SonarCloud quality checks to use safer xml parser.

https://sonarcloud.io/organizations/apache/rules?open=java%3AS2755&rule_key=java%3AS2755

### Types of changes

- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Bug fix (non-breaking change which fixes an issue)
- [x] Enhancement (improves an existing feature and functionality)
- [ ] Cleanup (Code refactoring and cleanup, that may add test cases)

### CODE REVIEW

**Comment by rohityadavcloud** [comment]
@blueorangutan package

**Comment by rohityadavcloud** [comment]
@blueorangutan test matrix

File: utils/src/test/java/org/apache/cloudstack/utils/security/ParserUtilsTest.java [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+
+import junit.framework.TestCase;
+
+public class ParserUtilsTest extends TestCase {
+
+    public void testGetSaferDocumentBuilderFactory() throws ParserConfigurationException {
+        final DocumentBuilderFactory factory = ParserUtils.getSaferDocumentBuilderFactory();
+        assertTrue(factory.getFeature(XMLConstants.FEATURE_SECURE_PROCESSING));
+        assertTrue(factory.getFeature("http://apache.org/xml/features/disallow-doctype-decl"));
+        assertFalse(factory.getFeature("http://xml.org/sax/features/external-general-entities"));
```
Comment (sureshanaparti) [inline utils/src/test/java/org/apache/cloudstack/utils/security/ParserUtilsTest.java:37]: can define / use constants for these urls?

**Review by sureshanaparti — APPROVED** [review]
code LGTM

**Comment by rohityadavcloud** [comment]
@blueorangutan test

**Comment by rohityadavcloud** [comment]
@blueorangutan test centos8 vmware-70u3

**Comment by rohityadavcloud** [comment]
@blueorangutan test rocky8 vmware-70u3

**Comment by rohityadavcloud** [comment]
@blueorangutan test ubuntu20 xcpng82

**Comment by rohityadavcloud** [comment]
@blueorangutan test ubuntu20 xcpng82

**Comment by shwstppr** [comment]
> Trillian test result (tid-4535) Environment: xcpng82 (x2), Advanced Networking with Mgmt server u20 Total time taken: 50774 seconds Marvin logs: https://github.com/blueorangutan/acs-prs/releases/download/trillian/pr6562-t4535-xcpng82.zip Smoke tests completed. 97 look OK, 3 have errors Only failed tests results shown below:
> Test 	Result 	Time (s) 	Test File
> test_attach_and_distribute_multiple_volumes 	`Error` 	16.97 	test_attach_multiple_volumes.py
> test_attach_multiple_volumes 	`Failure` 	13.88 	test_attach_multiple_volumes.py
> test_08_upgrade_kubernetes_ha_cluster 	`Failure` 	727.28 	test_kubernetes_clusters.py
> test_12_resize_volume_with_only_size_parameter 	`Error` 	1.08 	test_volumes.py

volumes error not related, https://github.com/apache/cloudstack/pull/6549 should fix it

**Review by shwstppr — APPROVED** [review]

**Review by harikrishna-patnala — APPROVED** [review]
LGTM

**Pull request merged**